azure-mgmt-monitor/src/main/java/com/microsoft/azure/management/monitor/implementation/AutoscaleProfileImpl.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.monitor.implementation;
+
+import com.microsoft.azure.management.apigeneration.LangDefinition;
+import com.microsoft.azure.management.monitor.AutoscaleNotification;
+import com.microsoft.azure.management.monitor.AutoscaleProfile;
+import com.microsoft.azure.management.monitor.AutoscaleSetting;
+import com.microsoft.azure.management.monitor.Recurrence;
+import com.microsoft.azure.management.monitor.RecurrenceFrequency;
+import com.microsoft.azure.management.monitor.ScaleCapacity;
+import com.microsoft.azure.management.monitor.ScaleRule;
+import com.microsoft.azure.management.monitor.TimeWindow;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.ExternalChildResourceImpl;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.GroupableResourceImpl;
+import org.joda.time.DateTime;
+import rx.Observable;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation for CdnProfile.
+ */
+@LangDefinition
+class AutoscaleProfileImpl
+        implements
+            AutoscaleProfile,
+            AutoscaleProfile.Definition,
```

Comment:
You were able to make enable profile definition as part of parent update without UpdateDefintionStages - this is because AutoscaleProfile.Definition<ParentT> is generic type but used as raw type when AutoscaleProfileImpl implements it. Usage of raw type was something we try to avoid.